lib/Minz/Request.php

```diff
@@ -261,6 +261,50 @@ public static function serverIsPublic($address) {
 		return (bool)$is_public;
 	}
 
+	private static function requestId() {
+		if (empty($_GET['rid']) || !ctype_xdigit($_GET['rid'])) {
+			$_GET['rid'] = uniqid();
+		}
+		return $_GET['rid'];
+	}
+
+	private static function setNotification($type, $content) {
+		Minz_Session::_param('notification', [ 'type' => $type, 'content' => $content ]);
+	}
+
+	public static function setGoodNotification($content) {
+		self::setNotification('good', $content);
+	}
+
+	public static function setBadNotification($content) {
+		self::setNotification('bad', $content);
+	}
+
+	public static function getNotification() {
+		$notif = null;
+
+		//Restore forwarded notifications
+		//TODO: Will need to ensure non-concurrency when landing https://github.com/FreshRSS/FreshRSS/pull/3096
```

Comment:
This comment would be more useful directly in the https://github.com/FreshRSS/FreshRSS/pull/3096 thread

Reply:
I will work on https://github.com/FreshRSS/FreshRSS/pull/3096 right after this PR lands, so it does not matter but was mainly intended for the current PR review